crates/astria-bridge-withdrawer/src/bridge_withdrawer/submitter/signer/frost_signer.rs

```diff
@@ -0,0 +1,312 @@
+
+        Ok(transaction)
+    }
+
+    pub(crate) fn address(&self) -> &Address {
+        &self.address
+    }
+}
+
+struct Round1Results {
+    responses: Vec<Round1Response>,
+    signing_package_commitments: BTreeMap<Identifier, round1::SigningCommitments>,
+}
+
+struct Round1Response {
+    id: Identifier,
+    commitment: axum::body::Bytes,
+    request_identifier: u32,
+}
+
+impl FrostSigner {
+    async fn execute_round_1(&self) -> Round1Results {
+        let stream = futures::stream::FuturesUnordered::new();
+        for (id, client) in &self.participant_clients {
+            let mut client = client.clone();
+            stream.push(async move {
+                let resp = client
+                    .execute_round_one(RoundOneRequest {})
+                    .await
+                    .wrap_err(format!(
+                        "failed to get part 1 response for participant with id {id:?}"
+                    ))?;
+                Ok((id, resp.into_inner()))
+            });
+        }
+        let results: Vec<eyre::Result<_>> = stream.collect::<Vec<_>>().await;
+        let mut signing_package_commitments: BTreeMap<Identifier, round1::SigningCommitments> =
+            BTreeMap::new();
+
+        let responses = results
+            .into_iter()
+            .filter_map(|res| match res {
+                Ok((id, part1)) => {
+                    let signing_commitment =
+                        round1::SigningCommitments::deserialize(&part1.commitment).ok()?;
+                    signing_package_commitments.insert(*id, signing_commitment);
+                    Some(Round1Response {
+                        id: *id,
+                        commitment: part1.commitment,
+                        request_identifier: part1.request_identifier,
+                    })
+                }
+                Err(_) => None,
+            })
+            .collect::<Vec<_>>();
+        Round1Results {
+            responses,
+            signing_package_commitments,
+        }
+    }
+
+    async fn execute_round_2(
+        &self,
+        responses: Vec<Round1Response>,
+        tx_bytes: Vec<u8>,
+    ) -> BTreeMap<Identifier, frost_ed25519::round2::SignatureShare> {
+        let stream = futures::stream::FuturesUnordered::new();
+        let request_commitments: Vec<CommitmentWithIdentifier> = responses
+            .iter()
+            .map(
+                |Round1Response {
+                     id,
+                     commitment,
+                     ..
+                 }| CommitmentWithIdentifier {
+                    commitment: commitment.clone(),
+                    participant_identifier: id.serialize().into(),
+                },
+            )
+            .collect();
+        for Round1Response {
+            id,
+            request_identifier,
+            ..
+        } in responses
+        {
+            let mut client = self
+                .participant_clients
+                .get(&id)
+                .expect(
```

Comment:
I mean - yeah, that doesn't seem wrong if you know exactly what round-1 does, but I think we should add a `wrap_err` and slap a big `BUG` in the message.